internal/xds/rbac/rbac_engine_test.go

```diff
@@ -271,106 +414,112 @@ func (s) TestRBACEngineConstruction(t *testing.T) {
 	}
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
-			if _, err := NewEngine(test.rbacConfig); (err != nil) != test.wantErr {
-				t.Fatalf("NewEngine(%+v) returned err: %v, wantErr: %v", test.rbacConfig, err, test.wantErr)
+			if _, err := NewChainEngine(test.rbacConfigs); (err != nil) != test.wantErr {
+				t.Fatalf("NewChainEngine(%+v) returned err: %v, wantErr: %v", test.rbacConfigs, err, test.wantErr)
 			}
 		})
 	}
 }
 
-// TestRBACEngine tests the RBAC Engine by configuring the engine in different
-// scenarios. After configuring the engine in a certain way, this test pings the
-// engine with different kinds of data representing incoming RPC's, and verifies
-// that it works as expected.
-func (s) TestRBACEngine(t *testing.T) {
+// TestChainEngine tests the chain of RBAC Engines by configuring the chain of
+// engines in a certain way in different scenarios. After configuring the chain
+// of engines in a certain way, this test pings the chain of engines with
+// different types of data representing incoming RPC's (piped into a context),
+// and verifies that it works as expected.
+func (s) TestChainEngine(t *testing.T) {
 	tests := []struct {
 		name        string
-		rbacConfig  *v3rbacpb.RBAC
+		rbacConfigs []*v3rbacpb.RBAC
 		rbacQueries []struct {
-			rpcData                *RPCData
-			wantMatchingPolicyName string
-			wantMatch              bool
+			rpcData        *rpcData
+			wantStatusCode codes.Code
 		}
 	}{
-		// TestSuccessCaseAnyMatch tests an RBAC Engine instantiated with a
-		// config with a policy with any rules for both permissions and
+		// TestSuccessCaseAnyMatch tests a single RBAC Engine instantiated with
```

Comment:
Many of the test cases here are similar to the ones in the above test. Please try to avoid testing the same thing repeatedly.

Another thing to consider is to define some of the protos used by both tests as globals. That way, the test table will be shorter and easier to read.